### PR TITLE
SMILES generation improvements

### PR DESCRIPTION
**Pull request by SJShaw**
Uses module modification domains to alter SMILES generation, adding methyl groups when possible for relevant methylation domains.

Includes a refactor of the SMILES logic to pass the relevant data in and removes some inaccurate edge cases at the same time.

### CODE REVIEW

File: antismash/modules/nrps_pks/test/test_smiles_generator.py [hunk 1]
```diff
@@ -37,3 +42,112 @@ def test_special_cases(self):
...
+    def test_double_bond_same_level(self):
+        bonds = get_bond_counts(Bonds("C=C"))
+        assert bonds == [2, 2]
+
+    def test_ring(self):
+        bonds = get_bond_counts(Bonds("C1NNC1"))
+        #                C1 N  N  C 1
+        assert bonds == [2, 1, 1, 2]
+
+    def test_carbon_rings(self):
```
Comment (kblin) [inline antismash/modules/nrps_pks/test/test_smiles_generator.py]: aromatic carbon rings, `C1CCCCC1` is also a carbon ring :wink:

File: antismash/modules/nrps_pks/smiles_generator.py [hunk 2]
```diff
@@ -163,6 +163,30 @@ def __iter__(self) -> Iterator[Atom]:
                 yield item
 
 
+def methylate(smiles: str, variant: str) -> str:
+    """ Add a methl group to the given smiles at the first available position.
```
Comment (kblin) [inline antismash/modules/nrps_pks/smiles_generator.py]: methl -> methyl

File: antismash/modules/nrps_pks/smiles_generator.py [hunk 3]
```diff
@@ -163,6 +163,30 @@ def __iter__(self) -> Iterator[Atom]:
...
+        Methylations of the first and last atom will not happen.
```
Comment (kblin) [inline antismash/modules/nrps_pks/smiles_generator.py]: Maybe mention that this is because we want to only methylate on the side-chain

File: antismash/modules/nrps_pks/test/test_smiles_generator.py [hunk 4]
```diff
@@ -151,3 +152,28 @@ def test_ring(self):
     def test_carbon_rings(self):
         bonds = get_bond_counts(Bonds("c1cccc1"))
         assert bonds == [0, 0, 0, 0, 0]
+
+
+class TestMethylation:
+    def test_c(self):
+        front = "NC(C(=O)C)"
+        back = "C(=O)O"
+        assert methylate(front + back, "C") == front + "(C)" + back
```
Comment (kblin) [inline antismash/modules/nrps_pks/test/test_smiles_generator.py]: This looks wrong, we're not methylating the side-chain here.
`methylate("NC(C(=O)C)C(=O)O", "C")` returns `NC(C(=O)C)(C)C(=O)O`, which has a second side-chain at the back-bone carbon. I'd expect `NC(C(=O)C(C))C(=O)O` instead.
